pkg/test/framework/components/ingress/kube.go

```diff
@@ -177,7 +177,11 @@ func getHTTPSAddressInner(env *kube.Environment, ns string) (interface{}, bool,
 func newKube(ctx resource.Context, cfg Config) Instance {
 	c := &kubeComponent{}
 	c.id = ctx.TrackResource(c)
-	c.namespace = cfg.Istio.Settings().IngressNamespace
+	if cfg.Namespace != nil {
+		c.namespace = cfg.Namespace.Name()
+	} else {
+		c.namespace = cfg.Istio.Settings().IngressNamespace
```

Comment:
How are you deploying the gateway to a different namespace and should that be baked into the istio deploy? 
If you're disabling the default ingress gateway I'd imagine something like this would work:

```go
framework.NewSuite(m).
  Setup(istio.Setup(&ist, func (cfg *istio.Config) {
  cfg.IngressNamespace = "my-custom-namespace"
  cfg.ControlPlaneValues = `
components:
  ingressGateways:
  - enabled: false`
  })).
```

Reply:
Yes this works @stevenctl  however, that namespace need to exist in the first place, secondly possible use case of using ingress is with echoboot instances and we can only create new namespaces with prefixes not with exact matches

Reply:
You could create the namespace in an earlier `Setup` at the suite scope and store a ref in a package level variable then just assign `ns.Name()` to get the generated name.

Reply:
Got your concern. Yes there's a WIP but let me see if I can get by without using custom ingress setup